core/src/calls.rs

```diff
@@ -11,18 +11,42 @@ impl<T: Metadata> Metadata for Option<T> {}
 impl<T: Metadata> Metadata for Box<T> {}
 impl<T: Sync + Send + 'static> Metadata for Arc<T> {}
 
+/// A future-conversion trait.
+pub trait WrapFuture<T, E> {
```

Comment:
Why not just call it `IntoFuture`? To avoid confusion with the erstwhile `futures` trait?

Reply:
Yeah, I want to avoid confusion with both `futures01::IntoFuture` trait and the `futures03::IntoFuture` struct. I think long-term we should consider `jsonrpc-derive` to require the methods to be `async` and then this issue is no longer a thing.

Reply:
Added an issue: https://github.com/paritytech/jsonrpc/issues/571